src/kv_manager/value.rs

```diff
@@ -0,0 +1,115 @@
+use serde::{Deserialize, Serialize};
+use std::{convert::TryFrom, path::PathBuf};
+
+use crate::{
+    encrypted_sled::Password,
+    gg20::types::{Entropy, PartyInfo},
+    mnemonic::FileIo,
+};
+type SigningKey = Vec<u8>;
+
+use super::{
+    error::{KvError, KvResult},
+    kv::Kv,
+};
+
+/// Kv manager for grpc services
+// pub type KvManager = Kv<KvValue>;
+#[derive(Clone)]
+pub struct KvManager {
+    kv: Kv<KvValue>,
+    io: FileIo,
+}
+
+impl KvManager {
+    pub fn new(root: &str, password: Password) -> KvResult<Self> {
+        Ok(KvManager {
+            kv: Kv::<KvValue>::new(root, password)?,
+            io: FileIo::new(PathBuf::from(root)),
+        })
+    }
+    pub fn kv(&self) -> &Kv<KvValue> {
+        &self.kv
+    }
+    pub fn io(&self) -> &FileIo {
+        &self.io
+    }
+}
+
+/// Enumerate the possible value types of the kv store
+// TODO: this is starting to get out of hand. Use better ways to address different types in the kv store
```

Comment:
Agreed.  It's fine for now but we should discuss a proper solution sometime.  Please post a github issue if you don't already have one.

Reply:
created a new issue. Will link to this lines of code when branch merged.